src/mpi/romio/adio/common/ad_close.c

```diff
@@ -61,7 +61,7 @@ void ADIO_Close(ADIO_File fd, int *error_code)
     }
 
     if (fd->hints) ADIOI_Free(fd->hints->ranklist);
-    if (fd->hints) ADIOI_Free(fd->hints->cb_config_list);
+    if (fd->hints && fd->hints->cb_config_list) ADIOI_Free(fd->hints->cb_config_list);
```

Comment:
I still contend that if these pointers are properly managed, there is no need to check them before free, but we can address that in a later commit.